src/xcode/ENA/ENA/Source/Services/HealthCertificate/HealthCertificateService.swift

```diff
@@ -537,56 +537,58 @@ class HealthCertificateService {
 	func updateValidityStatesAndNotifications(shouldScheduleTimer: Bool = true) {
 		Log.info("Update validity state and notifications.")
 
-		let currentAppConfiguration = appConfiguration.currentAppConfig.value
-		healthCertifiedPersons.forEach { healthCertifiedPerson in
-			healthCertifiedPerson.healthCertificates.forEach { healthCertificate in
-				let expirationThresholdInDays = currentAppConfiguration.dgcParameters.expirationThresholdInDays
-				let expiringSoonDate = Calendar.current.date(
-					byAdding: .day,
-					value: -Int(expirationThresholdInDays),
-					to: healthCertificate.expirationDate
-				)
+		DispatchQueue.global(qos: .default).async {
```

Comment:
```suggestion
		DispatchQueue.global(qos: .default).async { [weak self] in
```
otherwise we capture self in the closure as a strong reference

Reply:
In this case that shouldn't be a big downside, as the closure will be executed and will not be stored afterwards, so the strong reference will only be kept alive until the executions has finished. But just to make sure and not unnecessarily execute it I have added the weak reference now :)